iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -187,6 +187,146 @@ struct SliceOpConversion : public OpConversionPattern<mhlo::SliceOp> {
 };
 }  // namespace
 
+//===----------------------------------------------------------------------===//
+// mhlo.conv conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+bool isDepthwiseConv(mhlo::ConvOp op) {
+  auto shape = op.rhs().getType().cast<ShapedType>().getShape();
+  auto numGroups =
+      shape[op.dimension_numbers().kernel_input_feature_dimension().getInt()];
+  return op.feature_group_count() > 1u && op.feature_group_count() == numGroups;
+}
+
+/// Converts mhlo.conv operation to linalg named op.
+struct ConvOpConversion : public OpConversionPattern<mhlo::ConvOp> {
```

Comment:
Why here instead of in MHLO repo?

Reply:
I plan to upstream it after moving this to tensors world. Otherwise, it adds build cop's workload.